core/shared/src/main/scala/zio/ZRef.scala

```diff
@@ -307,9 +307,32 @@ object ZRef extends Serializable {
     def set(a: A): zio.IO[EA, Unit] = ZIO.fromEither(setter(Right(a))).flatMap(a0 => original.set(a0).mapError(fea))
 
     def setAsync(a: A): zio.IO[EA, Unit] = ZIO.fromEither(setter(Right(a))).map(a0 => original.setAsync(a0)).unit
-    // def maybeUpdate[E](f: B => Either[E,A])(implicit ea: EA <:< E, eb: EB <:< E): IO[E,B] = original.map { o =>
+    def maybeUpdate[E](f: B => Either[E, A])(implicit evea: EA <:< E, eveb: EB <:< E): IO[E, B] =
```

Comment:
@regiskuckaertz , I am having a hard time  here, would appreciate any help to get me unstuck

Reply:
@dsebban sorry quite busy right now, will have a look asap 😇